### DISCUSSION TITLE
error 'missing key {root_node_ID}'

### DISCUSSION THREAD

**Discussion by joachim-wesenberg** [freeplane/freeplane#270] [Help]
Background: I was a FreeMind user for many years. Since the support has been dropped I decided to switch to Freeplane. Mindmaps are **NOT** created using the UI, instead mm-files are being generated out of a legacy application from a database. The generated XML-map is  displayed/presented as expected  by Freeplane. 

Question 1: However, it throws one error which I am not able to figure out: 'missing key {showing the root_node_ID}'. Could you please point me to any XML-phrase where I could have a closer look at?

Question 2: Each node in the generated XML-map (and there might be thousends of it) has its own attribute attached. The attribute is shown automatically as soon as the mouse hovers over the node. Is there any XML-phrase which avoids this behaviour (no action when hovering over a node) ?  If necessary attribute-display can be forced by using the right-button-menu.

Thank you very much in advance - your help is very much appreciated.
Regards - Jo

**Comment by EdoFro** [comment]
hi,
### to **question 2:**

 change preferences at `Preferences…->Behaviour->Tooltips->Display tool tips`

**good to know**: you can **search** for any preference or command by using hotkey **Ctrl + F1** and doble clicking it

other **menu comands** that may be useful:

`View->Node attributes->Hide all attributes`

`Edit->Node properties->Attribute manager…`

----
### to question 1:

maybe you can attach an example XML so others can look at it and look for a solution

**An idea:**

- create a new map and save it right away
- compare its XML with yours and look at the differences for the root node and map characteristics

**An other idea:**

maybe in your XML there is no ID attribute:

look [in this template mindmap](https://github.com/freeplane/freeplane/blob/1.9.x/freeplane/src/external/resources/templates/standard-1.6.mm#:~:text=new_mindmap%22%20FOLDED%3D%22false%22-,ID%3D%22ID_696401721%22,-CREATED%3D%221610381621824%22%20MODIFIED) to see how the node has an "ID" in it

hope it helps

edo

**Comment by dpolivaev** [comment]
Yeah, could you please attach an example of the generated mind map file?

**Comment by joachim-wesenberg** [comment]
Thank you for your very fast reaction. Attached you will find an example which is content-wise meaningless but will reproduce the error.
Regards - Jo
[reproduce_error_MissingKey.txt](https://github.com/freeplane/freeplane/files/8332206/reproduce_error_MissingKey.txt)

**Comment by dpolivaev** [comment]
The map contains two errors: 
1. it uses attribute LOCALIZED_TEXT="mod_roi_it_infrastructure" in the root node in line 3. It leads to the error message because you need a translation. Replace LOCALIZED_TEXT by TEXT to fix it
2. the root style node element in line 7 should be closed at the end the style map. Here is the fix:
[reproduce_error_MissingKey-fixed.txt](https://github.com/freeplane/freeplane/files/8332388/reproduce_error_MissingKey-fixed.txt)

**Comment by joachim-wesenberg** [comment]
Dimitry,
wow, that was very quick. Your fix cured the problem 'missing key'. Thanx a lot. May I ask for a comment/XML-fix for question 2 (Attributes are hidden initially [works fine] and should still be hidden - even when hovering the mouse over a node).

**Comment by joachim-wesenberg** [comment]
EdoFro,
Thank you for your suggestions. You are quiet right  in solving the 'hide attribute'-problem via UI. However, I'm bound to solve that via XML only.The question is how to translate UI-functionality into XML-syntax.
Thanx again and best regards - Jo

**Reply by EdoFro** [reply]
I don't think it's possible.

Maybe you can programatically change the **user properties file** '**_auto.properties_**' that is in the **user directory**, 
so it contains `show_node_tooltips=false`

**Reply by EdoFro** [reply]
I still consider that 

> change preferences at `Preferences…->Behaviour->Tooltips->Display tool tips`

is the best option

**Reply by dpolivaev** [reply]
Agreed. This option is not saved in a mind map itself, it is only available as a user preference. If you want to have it change you can start a discussion topic and share your considerations.

**Reply by joachim-wesenberg** [reply]
Dimitry & EdoFro,
thanks again for your help. I will try to find a pragmatically way to get around the attribute issue.
Regards - Jo
